openpgp-dsm/src/lib.rs

```diff
@@ -1847,15 +1848,15 @@ fn api_curve_from_sequoia_curve(curve: SequoiaCurve) -> Result<ApiCurve> {
     }
 }
 
-fn try_unlock_p12(cert_file: String) -> Result<Identity> {
+fn try_unlock_p12(cert_file: String, passphrase: Option<&str>) -> Result<Identity> {
     let mut cert_stream = File::open(cert_file.clone())
         .context(format!("opening {}", cert_file))?;
     let mut cert = Vec::new();
     cert_stream.read_to_end(&mut cert)
         .context(format!("reading {}", cert_file))?;
-    // Try to unlock certificate without password first
+    // Try to parse certificate with passed or empty password first
     let mut first = true;
-    if let Ok(id) = Identity::from_pkcs12(&cert, "") {
+    if let Ok(id) = Identity::from_pkcs12(&cert, passphrase.unwrap_or("")) {
```

Comment:
E.g. try with the passed password or the empty password, and then keep trying other things such as env variables or user prompt.